src/components/EditSections/EditExtendedInfo/CreateResetPasswordControl/CreateResetPasswordControl.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash';
+
+import {
+  Col,
+  KeyValue,
+} from '@folio/stripes/components';
+
+import CreatePasswordModal from './Modals/CreatePasswordModal';
+import ResetPasswordModal from './Modals/ResetPasswordModal';
+
+import css from './CreateResetPasswordControl.css';
+
+class CreateResetPasswordControl extends React.Component {
+  static propTypes = {
+    email: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    userId: PropTypes.string.isRequired,
+    resources: PropTypes.shape({
+      isLocalPasswordSet: PropTypes.shape({
+        records: PropTypes.object,
+      }).isRequired,
+    }).isRequired,
+    mutator: PropTypes.shape({
+      resetPassword: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
+
+  static manifest = Object.freeze({
+    resetPassword: {
+      type: 'okapi',
+      path: 'bl-users/password-reset/link',
+      fetch: false,
+      throwErrors: false,
+    },
+    isLocalPasswordSet: {
+      type: 'okapi',
+      path: 'authn/credentials-existence?userId=!{userId}',
+    },
+  });
+
+  state = {
+    showModal: false,
+    link: '',
+  };
+
+  closeModal = () => {
+    this.setState({ showModal: false });
+  };
+
+  handleSuccessResponse = ({ link }) => {
+    this.setState({
+      link,
+      showModal: true,
+    });
+  };
+
+  handleLinkClick = () => {
+    const {
+      mutator: {
+        resetPassword,
+      },
+      userId,
+    } = this.props;
+
+    resetPassword
+      .POST({ userId })
+      .then(this.handleSuccessResponse)
+      .catch(this.closeModal);
+  };
+
+  createPasswordModal = () => {
+    return (
+      <CreatePasswordModal
+        isOpen={this.state.showModal}
```

Comment:
Please, consider the possibility to use state and props destructuring

Reply:
Do we have a rule always use destructuring?

Reply:
As I know destructuring is more preferable. Also I can't find reason to not to use it.(sure it will be more code but in my opinion destructuring brings clarity)